### DISCUSSION TITLE
Increased precision of BAT coordinates

### DISCUSSION THREAD

**Discussion by Jcruz161** [MDAnalysis/mdanalysis#4601] [Q&A]
Hello, I'm currently working on a project that requires the internal coordinates of a molecule, and while the BAT module seems to do exactly what I need, I've run into a bit of an issue regarding precision. The results are being returned as 32-bit floats and I really need them to be 64-bit doubles. 

I've been trying to read through the code to see if there's a flag or something I missed, but I can't find anything. Nor can I find any reason why the results would be returned as 32 bit values.

I did find this comment in one of the files: "Due to the use of Instruction Set Architecture (`ISA`) specific SIMD intrinsics in distopia via `VCL2`, the precision of your results may depend on the ISA available on your machine", and the discussion topic #3915, but they're both kind of over my head, and I'm not really sure how to use this information to fix my problem. 

Is there some simple fix here that I'm missing?

**Comment by orbeckst** [comment]
You can cast the float32 to float64 yourself (`numpy.asarray(coordinates, dtype=numpy.float64)`), but of course that won't gain you actual precision. I dont't think that we currently have a way to internally switch from float32 to float64 — better control over precision is on the [3.0 roadmap](https://www.mdanalysis.org/2023/10/25/towards_3.0/#enhancements) but maybe @hmacdope can provide more details as it's his domain.

**Comment by hmacdope** [comment]
As @orbeckst says, there is a distinction between precision (width of representation in bits) and "actual precision", i.e how many sig fig  (or ULP to be more technically correct) accurately represent your data correctly. 

My main question is why do you need them to be doubles? Do they need to be fed into something that only accepts a 64 bit representation from a code perspective or do they **actually** need to be precise enough that a 32 bit float will not cover them.  Perhaps understanding your use case better will allow me to make a more informed judgement. :smile: 

For clarity RE distopia, the comment "Due to the use of Instruction Set Architecture (ISA) specific SIMD intrinsics in distopia via VCL2, the precision of your results may depend on the ISA available on your machine" only applies if you use the `backend=distopia` flag in your distance calculations. Long story short using explicit SIMD intrinsics can change the  numerical accuracy of your results depending on what types of specialised registers you have available to work with (e.g if you have fused multiply add registers in your computer), but you don't really need to worry about this.
